src/components/common/CollabAnnoGeneSearch.jsx

```diff
@@ -0,0 +1,133 @@
+import _ from 'lodash'
+import React from 'react'
+import { Search, Input, Dropdown, Select, Icon, Segment } from 'semantic-ui-react'
+import { BackendAPI } from './API'
+import ReactGA from 'react-ga';
+
+export default class CollabAnnoGeneSearch extends React.Component {
+	
+	constructor(props) {
+		super(props);
+		this.state = {
+			isLoading: false,
+			results: [],
+			selection: null,
+			type: "gene",
+			searchQuery: "",
+			options: [],
+			value: []
+		};
+		this.call = null;
+	}
+
+	render() {
+
+		const { isLoading, results, options, type } = this.state;
+
+		return (
+				<Dropdown
+					className="collab-anno-drop"
+					// category
+					multiple
+					search
+					selection
+					loading={isLoading}
+					deburr={true}
+					closeOnChange={true}
+					onSearchChange={(evt, input) => {
+						evt.persist();
+						this.setState({searchQuery: evt.target.value});
+						this.stopRequest();
+						this.handleSearchChangeDebounced.cancel();
+						this.handleSearchChangeDebounced(evt, input);
+					}}
+					onChange={this.updateValues.bind(this)}
+					stopRequest={() => {
+						if (this.call != null) this.call.end();
+					}}
+					options={options}
+					results={results}
+					value={this.state.value}
+					searchQuery={this.state.searchQuery}
+					type={type}
+				/>
+		);
+
+	}
+
+	updateValues(e, { value }) {
+		this.setState({
+			'value': value,
+			'results': value.map((v) => {return {'key': v, 'text': v, 'value':v}}),
```

Comment:
So by default the Dropdown implementation only shows results that are selected which are in the list of possible options (and results). We're using this dropdown with the gene search function of scope, to allow people to select multiple genes. The problem is, that we only populate the results of the dropdown with the relevant results from their search, removing the old ones. 

This code is there to maintain a copy of the selected genes within the results and options, so that it will always appear as selected. Without this, once someone changes their search (and the results/options are cleared) their currently selected genes will dissapear.

Reply:
OK, I see the big picture. Had a dig into the (aweful) `react-semantic-ui` docs and your usage here is clear now.